aten/src/ATen/function_wrapper.py

```diff
@@ -732,6 +742,23 @@ def process_option(option, output_options):
             top_env['type_method_definitions'].append(
                 TYPE_METHOD_DEFINITION_ABSTRACT.substitute(env))
 
+            # Set up type promotion, if used
+            unify_types = ''
+            has_side_effects = any('Tensor' in f and 'const' not in f
```

Comment:
can you post an in-place example?  Presumably we'd want to allow promotion of the non-inplace Tensor/Scalar arguments?

Reply:
That's true, but this is harder to handle in general, because you need to know which arg is the non-inplace one (can we assume it is the other than `self`? idk). I'd leave that for a future PR if you're ok with that